File: app/src/main/java/wseemann/media/romote/activity/ShakeActivity.java
package wseemann.media.romote.activity;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.wseemann.ecp.api.ResponseCallback;
import com.wseemann.ecp.core.KeyPressKeyValues;
import com.wseemann.ecp.request.KeyPressRequest;
import javax.inject.Inject;
import dagger.hilt.android.AndroidEntryPoint;
import timber.log.Timber;
import wseemann.media.romote.preferences.AppPreferences;
import wseemann.media.romote.utils.CommandHelper;
import wseemann.media.romote.utils.ShakeMonitor;
import wseemann.media.romote.utils.WindowInsetsUtils;

@AndroidEntryPoint
public class ShakeActivity extends AppCompatActivity {

    @Inject
    protected AppPreferences appPreferences;

    @Inject
    protected CommandHelper commandHelper;

    private ShakeMonitor mShakeMonitor;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        WindowInsetsUtils.enableRomoteEdgeToEdge(this);

        mShakeMonitor = new ShakeMonitor(this);
        mShakeMonitor.setOnShakeListener(mShakeListener);

        if (shakeEnabled()) {
            mShakeMonitor.resume();
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();

        if (shakeEnabled()) {
            mShakeMonitor.pause();
        }
    }

    private final ShakeMonitor.OnShakeListener mShakeListener = new ShakeMonitor.OnShakeListener() {
        @Override
        public void onShake() {
            String url = commandHelper.getDeviceURL();

            try {
                KeyPressRequest keyPressRequest = new KeyPressRequest(url, KeyPressKeyValues.PLAY.getValue());
                keyPressRequest.sendAsync(new ResponseCallback<>() {
                    @Override
                    public void onSuccess(@Nullable Void unused) {

                    }

                    @Override
                    public void onError(@NonNull Exception e) {

                    }
                });
            } catch (Exception ex) {
                Timber.e(ex, "Failed to execute command");
            }
        }
    };

    private boolean shakeEnabled() {
        return appPreferences.isShakeToPauseEnabled();
    }
}

File: app/src/main/java/wseemann/media/romote/activity/ShakeActivity.kt
package wseemann.media.romote.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.wseemann.ecp.core.KeyPressKeyValues
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.launch
import wseemann.media.romote.device.DeviceManager
import wseemann.media.romote.di.IoDispatcher
import wseemann.media.romote.preferences.AppPreferences
import wseemann.media.romote.utils.ShakeMonitor
import wseemann.media.romote.utils.enableRomoteEdgeToEdge
import javax.inject.Inject

@AndroidEntryPoint
open class ShakeActivity : AppCompatActivity() {

    @Inject
    @IoDispatcher
    lateinit var ioDispatcher: CoroutineDispatcher

    @Inject
    lateinit var appPreferences: AppPreferences

    @Inject
    lateinit var deviceManager: DeviceManager

    private lateinit var shakeMonitor: ShakeMonitor

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        enableRomoteEdgeToEdge(this)

        shakeMonitor = ShakeMonitor(this)
        shakeMonitor.setOnShakeListener { onShake() }

        if (shakeEnabled()) {
            shakeMonitor.resume()
        }
    }

    override fun onDestroy() {
        super.onDestroy()

        if (shakeEnabled()) {
            shakeMonitor.pause()
        }
    }

    private fun onShake() {
        lifecycleScope.launch(ioDispatcher) {
            deviceManager.getConnectedDevice()?.performKeyPress(KeyPressKeyValues.PLAY)
        }
    }

    private fun shakeEnabled(): Boolean = appPreferences.isShakeToPauseEnabled()
}
